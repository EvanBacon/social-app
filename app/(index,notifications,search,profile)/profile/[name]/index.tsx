export {ProfileScreen as default} from 'view/screens/Profile'
