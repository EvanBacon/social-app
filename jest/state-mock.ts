import {RootStoreModel} from './../src/state/models/root-store'
import {NavigationTabModel} from './../src/state/models/navigation'
import {SessionModel} from '../src/state/models/session'
import {NavigationModel} from '../src/state/models/navigation'
import {ShellUiModel} from '../src/state/models/shell-ui'
import {MeModel} from '../src/state/models/me'
import {OnboardModel} from '../src/state/models/onboard'
import {ProfilesViewModel} from '../src/state/models/profiles-view'
import {LinkMetasViewModel} from '../src/state/models/link-metas-view'
import {MembershipsViewModel} from '../src/state/models/memberships-view'
import {FeedModel} from '../src/state/models/feed-view'
import {NotificationsViewModel} from '../src/state/models/notifications-view'
import {LRUMap} from 'lru_map'

const mockedSessionStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  data: {
    service: '',
    refreshJwt: '',
    accessJwt: '',
    handle: '',
    did: '',
  },
  online: false,
  attemptingConnect: false,
  rootStore: {} as RootStoreModel,
  hasSession: true,
  clear: jest.fn(),
  setState: jest.fn(),
  setOnline: jest.fn(),
  updateAuthTokens: jest.fn(),
  connect: jest.fn(),
  describeService: jest.fn(),
  login: jest.fn(),
  createAccount: jest.fn(),
  logout: jest.fn(),

  // unknown added because of the missing private methods: _connectPromise, configureApi & _connect
} as unknown as SessionModel

const mockedNavigationTabStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  id: 0,
  history: [
    {
      url: '',
      ts: 0,
      title: '',
      id: 0,
    },
  ],
  index: 0,
  isNewTab: false,
  current: {
    url: '',
    ts: 0,
    title: '',
    id: 0,
  },
  canGoBack: false,
  canGoForward: false,
  backTen: [
    {
      url: '',
      title: '',
      index: 0,
      id: 0,
    },
  ],
  forwardTen: [
    {
      url: '',
      title: '',
      index: 0,
      id: 0,
    },
  ],
  navigate: jest.fn(),
  refresh: jest.fn(),
  goBack: jest.fn(),
  fixedTabReset: jest.fn(),
  goForward: jest.fn(),
  goToIndex: jest.fn(),
  setTitle: jest.fn(),
  setIsNewTab: jest.fn(),
  fixedTabPurpose: 0,
  getBackList: jest.fn(),
  getForwardList: jest.fn(),
} as NavigationTabModel

const mockedNavigationStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  tabs: [],
  tabIndex: 0,
  clear: jest.fn(),
  tab: mockedNavigationTabStore,
  tabCount: 1,
  isCurrentScreen: jest.fn(),
  navigate: jest.fn(),
  refresh: jest.fn(),
  setTitle: jest.fn(),
  handleLink: jest.fn(),
  switchTo: jest.fn(),
  setActiveTab: jest.fn(),
  closeTab: jest.fn(),
  newTab: jest.fn(),
} as NavigationModel

const mockedShellStore = {
  minimalShellMode: false,
  isMainMenuOpen: false,
  isModalActive: false,
  activeModal: undefined,
  isLightboxActive: false,
  activeLightbox: undefined,
  isComposerActive: false,
  composerOpts: undefined,
  setMainMenuOpen: jest.fn(),
  setMinimalShellMode: jest.fn(),
  openModal: jest.fn(),
  closeModal: jest.fn(),
  closeComposer: jest.fn(),
  closeLightbox: jest.fn(),
  openComposer: jest.fn(),
  openLightbox: jest.fn(),
} as ShellUiModel

const mockedMeStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  did: '',
  handle: '',
  displayName: '',
  description: '',
  avatar: '',
  notificationCount: 0,
  rootStore: {} as RootStoreModel,
  // TODO: mock these models
  memberships: {
    placeholder: null,
  } as MembershipsViewModel,
  mainFeed: {
    placeholder: null,
  } as FeedModel,
  notifications: {
    placeholder: null,
  } as NotificationsViewModel,
  clear: jest.fn(),
  load: jest.fn(),
  clearNotificationCount: jest.fn(),
  fetchStateUpdate: jest.fn(),
  refreshMemberships: jest.fn(),
} as MeModel

const mockedOnboardStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  isOnboarding: false,
  stage: '',
  start: jest.fn(),
  stop: jest.fn(),
  next: jest.fn(),
} as OnboardModel

const mockedProfileStore = {
  hydrate: jest.fn(),
  serialize: jest.fn(),
  cache: new LRUMap(100),
  rootStore: {} as RootStoreModel,
  getProfile: jest.fn(),
  overwrite: jest.fn(),
} as ProfilesViewModel

const mockedLinkMetasStore = {
  hydrate: jest.fn(),
  serialize: jest.fn(),
  cache: new LRUMap(100),
  rootStore: {} as RootStoreModel,
  getLinkMeta: jest.fn(),
} as LinkMetasViewModel

export const mockedRootStore = {
  api: false,
  resolveName: jest.fn(),
  serialize: jest.fn(),
  hydrate: jest.fn(),
  fetchStateUpdate: jest.fn(),
  clearAll: jest.fn(),
  session: mockedSessionStore,
  nav: mockedNavigationStore,
  shell: mockedShellStore,
  me: mockedMeStore,
  onboard: mockedOnboardStore,
  profiles: mockedProfileStore,
  linkMetas: mockedLinkMetasStore,
}
