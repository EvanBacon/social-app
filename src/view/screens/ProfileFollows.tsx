import React from 'react'
import {View} from 'react-native'
import {useFocusEffect} from '@react-navigation/native'
import {withAuthRequired} from 'view/com/auth/withAuthRequired'
import {ViewHeader} from '../com/util/ViewHeader'
import {ProfileFollows as ProfileFollowsComponent} from '../com/profile/ProfileFollows'
import {useStores} from 'state/index'
import {useLocalSearchParams} from 'expo-router'

export const ProfileFollowsScreen = withAuthRequired(() => {
  const store = useStores()
  const {name} = useLocalSearchParams<{name: string}>()

  useFocusEffect(
    React.useCallback(() => {
      store.shell.setMinimalShellMode(false)
    }, [store]),
  )

  return (
    <View>
      <ViewHeader title="Following" />
      <ProfileFollowsComponent name={name} />
    </View>
  )
})
