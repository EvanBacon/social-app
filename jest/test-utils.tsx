import React from 'react'
import {render} from '@testing-library/react-native'
import {GestureHandlerRootView} from 'react-native-gesture-handler'
import {RootSiblingParent} from 'react-native-root-siblings'
import {SafeAreaProvider} from 'react-native-safe-area-context'
import {RootStoreModel, RootStoreProvider} from '../src/state'
import {mockedRootStore} from './state-mock'

const WrappedComponent = ({children}: any) => {
  return (
    // eslint-disable-next-line react-native/no-inline-styles
    <GestureHandlerRootView style={{flex: 1}}>
      <RootSiblingParent>
        <RootStoreProvider value={mockedRootStore as RootStoreModel}>
          <SafeAreaProvider>{children}</SafeAreaProvider>
        </RootStoreProvider>
      </RootSiblingParent>
    </GestureHandlerRootView>
  )
}

const customRender = (ui: any, options?: any) =>
  render(ui, {wrapper: WrappedComponent, ...options})

// re-export everything
export * from '@testing-library/react-native'

// override render method
export {customRender as render}
