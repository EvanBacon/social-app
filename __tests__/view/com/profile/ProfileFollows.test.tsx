import React from 'react'
import {cleanup, fireEvent, render} from '../../../../jest/test-utils'
import {ProfileFollows} from '../../../../src/view/com/profile/ProfileFollows'
import {
  mockedLogStore,
  mockedUserFollowsStore,
} from '../../../../__mocks__/state-mock'

describe('ProfileFollows', () => {
  const mockedProps = {
    name: 'test user',
  }
  afterAll(() => {
    jest.clearAllMocks()
    cleanup()
  })

  it('renders activity indicator', async () => {
    jest
      .spyOn(React, 'useState')
      .mockReturnValue([mockedUserFollowsStore, jest.fn()])
    const {findByTestId} = render(<ProfileFollows name="test user 2" />)

    const profileFollowsActivityIndicatorView = await findByTestId(
      'profileFollowsActivityIndicatorView',
    )
    expect(profileFollowsActivityIndicatorView).toBeTruthy()
  })

  it('catches setup error', async () => {
    jest.spyOn(React, 'useState').mockReturnValue([
      {
        ...mockedUserFollowsStore,
        setup: jest.fn().mockRejectedValue(''),
        hasError: true,
      },
      jest.fn(),
    ])

    const {findByTestId} = render(<ProfileFollows {...mockedProps} />)
    expect(mockedLogStore.error).toHaveBeenCalled()

    const errorMessageTryAgainButton = await findByTestId(
      'errorMessageTryAgainButton',
    )
    expect(errorMessageTryAgainButton).toBeTruthy()
    fireEvent.press(errorMessageTryAgainButton)

    expect(mockedUserFollowsStore.refresh).toHaveBeenCalled()
  })

  it('renders list', async () => {
    jest
      .spyOn(React, 'useState')
      .mockReturnValue([mockedUserFollowsStore, jest.fn()])
    const {findByTestId} = render(<ProfileFollows {...mockedProps} />)

    const profileFollowsFlatList = await findByTestId('profileFollowsFlatList')
    expect(profileFollowsFlatList).toBeTruthy()
  })

  it('matches snapshot', () => {
    const page = render(<ProfileFollows {...mockedProps} />)
    expect(page).toMatchSnapshot()
  })
})
