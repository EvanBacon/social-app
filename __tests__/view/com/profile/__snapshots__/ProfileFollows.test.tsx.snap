// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileFollows matches snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    testID="profileFollowsFlatList"
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "_reactKey": "1",
            "declaration": Object {
              "actorType": "",
              "cid": "",
            },
            "did": "test did",
            "displayName": "test name",
            "handle": "testhandle",
            "indexedAt": "",
          },
          Object {
            "_reactKey": "2",
            "declaration": Object {
              "actorType": "",
              "cid": "",
            },
            "did": "test did2",
            "displayName": "test name 2",
            "handle": "testhandle2",
            "indexedAt": "",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      viewabilityConfigCallbackPairs={Array []}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "borderTopWidth": 1,
                },
                Object {
                  "backgroundColor": "#ffffff",
                },
                Object {
                  "borderColor": "#e4e2e2",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "paddingBottom": 10,
                    "paddingLeft": 10,
                    "paddingTop": 10,
                    "width": 60,
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight={40}
                  bbWidth={40}
                  focusable={false}
                  height={40}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "flex": 0,
                        "height": 40,
                        "width": 40,
                      },
                    ]
                  }
                  vbHeight={100}
                  vbWidth={100}
                  width={40}
                >
                  <RNSVGGroup>
                    <RNSVGDefs>
                      <RNSVGLinearGradient
                        gradient={
                          Array [
                            0,
                            -16742913,
                            1,
                            -14631929,
                          ]
                        }
                        gradientTransform={null}
                        gradientUnits={0}
                        name="grad"
                        x1="0"
                        x2="1"
                        y1="0"
                        y2="1"
                      />
                    </RNSVGDefs>
                    <RNSVGCircle
                      cx="50"
                      cy="50"
                      fill={
                        Array [
                          1,
                          "grad",
                        ]
                      }
                      propList={
                        Array [
                          "fill",
                        ]
                      }
                      r="50"
                    />
                    <RNSVGText
                      content={null}
                      dx={Array []}
                      dy={Array []}
                      fill={4294967295}
                      font={
                        Object {
                          "fontSize": "50",
                          "fontWeight": "bold",
                          "textAnchor": "middle",
                        }
                      }
                      propList={
                        Array [
                          "fill",
                        ]
                      }
                      rotate={Array []}
                      x={
                        Array [
                          "50",
                        ]
                      }
                      y={
                        Array [
                          "67",
                        ]
                      }
                    >
                      <RNSVGTSpan
                        content="T"
                        dx={Array []}
                        dy={Array []}
                        font={Object {}}
                        rotate={Array []}
                        x={Array []}
                        y={Array []}
                      />
                    </RNSVGText>
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                    "paddingBottom": 10,
                    "paddingRight": 10,
                    "paddingTop": 10,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                      },
                      Object {
                        "fontSize": 16,
                        "letterSpacing": 0.25,
                      },
                      Array [
                        Object {
                          "fontWeight": "bold",
                        },
                        Object {
                          "color": "#000000",
                        },
                      ],
                    ]
                  }
                >
                  test name
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                      },
                      Object {
                        "fontSize": 14,
                        "letterSpacing": 0.25,
                      },
                      Object {
                        "color": "#645454",
                      },
                    ]
                  }
                >
                  @
                  testhandle
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "borderTopWidth": 1,
                },
                Object {
                  "backgroundColor": "#ffffff",
                },
                Object {
                  "borderColor": "#e4e2e2",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "paddingBottom": 10,
                    "paddingLeft": 10,
                    "paddingTop": 10,
                    "width": 60,
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight={40}
                  bbWidth={40}
                  focusable={false}
                  height={40}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "flex": 0,
                        "height": 40,
                        "width": 40,
                      },
                    ]
                  }
                  vbHeight={100}
                  vbWidth={100}
                  width={40}
                >
                  <RNSVGGroup>
                    <RNSVGDefs>
                      <RNSVGLinearGradient
                        gradient={
                          Array [
                            0,
                            -2424577,
                            1,
                            -6862849,
                          ]
                        }
                        gradientTransform={null}
                        gradientUnits={0}
                        name="grad"
                        x1="0"
                        x2="1"
                        y1="0"
                        y2="1"
                      />
                    </RNSVGDefs>
                    <RNSVGCircle
                      cx="50"
                      cy="50"
                      fill={
                        Array [
                          1,
                          "grad",
                        ]
                      }
                      propList={
                        Array [
                          "fill",
                        ]
                      }
                      r="50"
                    />
                    <RNSVGText
                      content={null}
                      dx={Array []}
                      dy={Array []}
                      fill={4294967295}
                      font={
                        Object {
                          "fontSize": "50",
                          "fontWeight": "bold",
                          "textAnchor": "middle",
                        }
                      }
                      propList={
                        Array [
                          "fill",
                        ]
                      }
                      rotate={Array []}
                      x={
                        Array [
                          "50",
                        ]
                      }
                      y={
                        Array [
                          "67",
                        ]
                      }
                    >
                      <RNSVGTSpan
                        content="T"
                        dx={Array []}
                        dy={Array []}
                        font={Object {}}
                        rotate={Array []}
                        x={Array []}
                        y={Array []}
                      />
                    </RNSVGText>
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                    "paddingBottom": 10,
                    "paddingRight": 10,
                    "paddingTop": 10,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                      },
                      Object {
                        "fontSize": 16,
                        "letterSpacing": 0.25,
                      },
                      Array [
                        Object {
                          "fontWeight": "bold",
                        },
                        Object {
                          "color": "#000000",
                        },
                      ],
                    ]
                  }
                >
                  test name 2
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                      },
                      Object {
                        "fontSize": 14,
                        "letterSpacing": 0.25,
                      },
                      Object {
                        "color": "#645454",
                      },
                    ]
                  }
                >
                  @
                  testhandle2
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
